launch/launch/substitutions/boolean_substitution.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for boolean substitutions."""
+
+from typing import Iterable
+from typing import Text
+
+from .substitution_failure import SubstitutionFailure
+from ..conditions import evaluate_condition_expression
+from ..conditions import InvalidConditionExpressionError
+from ..frontend import expose_substitution
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitution import Substitution
+from ..utilities import normalize_to_list_of_substitutions
+from ..utilities import perform_substitutions
+
+
+@expose_substitution('not')
+class NotSubstitution(Substitution):
+    """Substitution that returns 'not' of the input boolean value."""
+
+    def __init__(self, value: SomeSubstitutionsType) -> None:
+        """Create a NotSubstitution substitution."""
+        super().__init__()
+
+        self.__value = normalize_to_list_of_substitutions(value)
+
+    @classmethod
+    def parse(cls, data: Iterable[SomeSubstitutionsType]):
+        """Parse `NotSubstitution` substitution."""
+        if len(data) != 1:
+            raise TypeError('not substitution expects 1 argument')
+        return cls, {'value': data[0]}
+
+    @property
+    def value(self) -> Substitution:
+        """Getter for value."""
+        return self.__value
+
+    def describe(self) -> Text:
+        """Return a description of this substitution as a string."""
+        return f'NotSubstitution({self.value})'
+
+    def perform(self, context: LaunchContext) -> Text:
+        """Perform the substitution."""
+        try:
+            condition_expression = normalize_to_list_of_substitutions(
+                perform_substitutions(context, self.value)
+            )
+            condition = evaluate_condition_expression(context, condition_expression)
+        except InvalidConditionExpressionError:
+            raise SubstitutionFailure(f'invalid condition expression: {self.value}')
+
+        return str(not condition)
```

Comment:
@kenji-miyake nit: here and below, though perhaps redundant, consider `lower()`-ing the outcome string for consistency

Reply:
Fixed in a223ec33ad2d349d2dba8a0dbabf5ae4d157215d.

Reply:
@kenji-miyake yeah, just to normalize what `launch` understands as a boolean literal.